### PR TITLE
Monkey sentience caps from TG

### PR DESCRIPTION
**Pull request by YouCanBeBetterThanThat**
# Document the changes in your pull request
ports https://github.com/tgstation/tgstation/pull/51621 from TG

# Wiki Documentation
document the crate about monkey sentience

# Changelog

:cl:  tralezab
rscadd: monkey sentience caps
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Monke

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -345,6 +345,92 @@
...
+	if(slot != ITEM_SLOT_HEAD)
+		return
+	if(!ismonkey(user) || user.ckey)
+		var/mob/living/something = user
+		to_chat(something, "<span class='boldnotice'>You feel a stabbing pain in the back of your head for a moment.</span>")
+		something.apply_damage(5,BRUTE,BODY_ZONE_HEAD,FALSE,FALSE,FALSE) //notably: no damage resist (it's in your helmet), no damage spread (it's in your helmet)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+		return
+	magnification = user //this polls ghosts
+	visible_message("<span class='warning'>[src] powers up!</span>")
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
	visible_message(span_warning("[src] powers up!"))
```

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -345,6 +345,92 @@
...
+	. += "<span class='warning'>GENETIC MAKEUP MASS SUSCEPTIBILITY</span>"
+	. += "<span class='boldnotice'>Ask your CMO if mind magnification is right for you.</span>"
+
+/obj/item/clothing/head/helmet/monkey_sentience/equipped(mob/user, slot)
+	. = ..()
+	if(slot != ITEM_SLOT_HEAD)
+		return
+	if(!ismonkey(user) || user.ckey)
+		var/mob/living/something = user
+		to_chat(something, "<span class='boldnotice'>You feel a stabbing pain in the back of your head for a moment.</span>")
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
		to_chat(something, span_boldnotice("You feel a stabbing pain in the back of your head for a moment."))
```

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -345,6 +345,92 @@
...
+		return
+	magnification = user //this polls ghosts
+	visible_message("<span class='warning'>[src] powers up!</span>")
+	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+	polling = TRUE
+	var/list/candidates = pollCandidatesForMob("Do you want to play as a mind magnified monkey?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, magnification, POLL_IGNORE_SENTIENCE_POTION)
+	polling = FALSE
+	if(!candidates.len)
+		magnification = null
+		visible_message("<span class='notice'>[src] falls silent. Maybe you should try again later?</span>")
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
		visible_message(span_notice("[src] falls silent. Maybe you should try again later?"))
```

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -345,6 +345,92 @@
...
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+	var/mob/picked = pick(candidates)
+	magnification.key = picked.key
+	playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
+	to_chat(magnification, "<span class='notice'>You're a mind magnified monkey! Protect your helmet with your life- if you lose it, your sentience goes with it!</span>")
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
	to_chat(magnification, span_notice("You're a mind magnified monkey! Protect your helmet with your life- if you lose it, your sentience goes with it!"))
```

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -345,6 +345,92 @@
...
+/obj/item/clothing/head/helmet/monkey_sentience/Destroy()
+	. = ..()
+	disconnect()
+
+/obj/item/clothing/head/helmet/monkey_sentience/proc/disconnect()
+	if(!magnification) //not put on a viable head
+		return
+	if(!polling)//put on a viable head, but taken off after polling finished.
+		if(magnification.client)
+			to_chat(magnification, "<span class='userdanger'>You feel your flicker of sentience ripped away from you, as everything becomes dim...</span>")
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
			to_chat(magnification, span_userdanger("You feel your flicker of sentience ripped away from you, as everything becomes dim..."))
```

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -345,6 +345,92 @@
...
+				if(2) //brain death
+					magnification.apply_damage(500,BRAIN,BODY_ZONE_HEAD,FALSE,FALSE,FALSE)
+				if(3) //primal gene (gorilla)
+					magnification.gorillize()
+				if(4) //genetic mass susceptibility (gib)
+					magnification.gib()
+	//either used up correctly or taken off before polling finished (punish this by destroying the helmet)
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+	playsound(src, "sparks", 100, TRUE)
+	visible_message("<span class='warning'>[src] fizzles and breaks apart!</span>")
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
	visible_message(span_warning("[src] fizzles and breaks apart!"))
```

File: code/modules/clothing/head/helmet.dm [hunk 7]
```diff
@@ -345,6 +345,92 @@
...
+	icon_state = "[icon_state][rand(1,3)]"
+
+/obj/item/clothing/head/helmet/monkey_sentience/examine(mob/user)
+	. = ..()
+	. += "<span class='boldwarning'>---WARNING: REMOVAL OF HELMET ON SUBJECT MAY LEAD TO:---</span>"
+	. += "<span class='warning'>BLOOD RAGE</span>"
+	. += "<span class='warning'>BRAIN DEATH</span>"
+	. += "<span class='warning'>PRIMAL GENE ACTIVATION</span>"
+	. += "<span class='warning'>GENETIC MAKEUP MASS SUSCEPTIBILITY</span>"
+	. += "<span class='boldnotice'>Ask your CMO if mind magnification is right for you.</span>"
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
	. += span_boldwarning("---WARNING: REMOVAL OF HELMET ON SUBJECT MAY LEAD TO:---")
	. += span_warning("BLOOD RAGE")
	. +=  span_warning("BRAIN DEATH")
	. += span_warning("PRIMAL GENE ACTIVATION")
	. +=  span_warning("GENETIC MAKEUP MASS SUSCEPTIBILITY")
	. += span_boldnotice("Ask your CMO if mind magnification is right for you.")
```

idk but you probavbly could do . += the entire thing and add <br> to the end and \ or \n\ to switch lines

**Review by nesisknesis — CHANGES_REQUESTED** [review]
Find an existing helmet and edit it to be the new sprite. ANYTHING would be better than the current sprite.

**Comment by YouCanBeBetterThanThat** [comment]
> Find an existing helmet and edit it to be the new sprite. ANYTHING would be better than the current sprite.

any suggestions for an existing one?

**Comment by ghost** [comment]
> > Find an existing helmet and edit it to be the new sprite. ANYTHING would be better than the current sprite.
> 
> any suggestions for an existing one?

Cockcult helmet!

File: code/modules/clothing/head/helmet.dm [hunk 8]
```diff
@@ -345,6 +345,92 @@
...
+
+	icon_state = "monkeymind"
+	item_state = "monkeymind"
+	strip_delay = 100
+	var/mob/living/carbon/monkey/magnification = null ///if the helmet is on a valid target (just works like a normal helmet if not (cargo please stop))
+	var/polling = FALSE///if the helmet is currently polling for targets (special code for removal)
+
+/obj/item/clothing/head/helmet/monkey_sentience/Initialize()
+	. = ..()
+	icon_state = "[icon_state][rand(1,3)]"
```
Comment (tattax) [inline code/modules/clothing/head/helmet.dm:362]: you set it here to rand(1,3) but checks shows you added only the 1st sprite of monkeymind

**Pull request closed without merging**